src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TreeViewTests.cs

```diff
@@ -0,0 +1,554 @@
+                            <ContentThemeTransition />
+                      </TransitionCollection>");
+					treeView.ItemContainerTransitions = transition;
+					var node = new TreeViewNode();
+					treeView.RootNodes.Add(node);
+					var listControl = FindVisualChildByName(treeView, "ListControl") as TreeViewList;
+					var treeViewItem = listControl.ContainerFromItem(node) as TreeViewItem;
+					Verify.AreEqual(treeViewItem.ContentTransitions, transition);
+				};
+			});
+		}
+
+		[TestMethod]
+		public void TreeViewItemsSourceTest()
+		{
+			RunOnUIThread.Execute(() =>
+			{
+				var treeView = new TreeView();
+				var items = CreateTreeViewItemsSource();
+				treeView.ItemsSource = items;
+
+				Verify.AreEqual(treeView.RootNodes.Count, 2);
+				Verify.AreEqual(treeView.RootNodes[0].Content as TreeViewItemSource, items[0]);
+			});
+		}
+
+		[TestMethod]
+		public void TreeViewItemsSourceUpdateTest()
+		{
+			RunOnUIThread.Execute(() =>
+			{
+				var treeView = new TreeView();
+				var items = CreateTreeViewItemsSource();
+				treeView.ItemsSource = items;
+
+				// Insert
+				var newItem = new TreeViewItemSource() { Content = "newItem" };
+				items.Add(newItem);
+				Verify.AreEqual(treeView.RootNodes.Count, 3);
+				var itemFromNode = treeView.RootNodes[2].Content as TreeViewItemSource;
+				Verify.AreEqual(newItem.Content, itemFromNode.Content);
+
+				// Remove
+				items.Remove(newItem);
+				Verify.AreEqual(treeView.RootNodes.Count, 2);
+
+				// Replace
+				var item3 = new TreeViewItemSource() { Content = "3" };
+				items[1] = item3;
+				itemFromNode = treeView.RootNodes[1].Content as TreeViewItemSource;
+				Verify.AreEqual(item3.Content, itemFromNode.Content);
+
+				// Clear
+				items.Clear();
+				Verify.AreEqual(treeView.RootNodes.Count, 0);
+
+			});
+		}
+
+		[TestMethod]
+		public void TreeViewNodeStringableTest()
+		{
+			RunOnUIThread.Execute(() =>
+			{
+				var node = new TreeViewNode() { Content = "Node" };
+				Verify.AreEqual(node.Content, node.ToString());
+
+				// Test inherited type
+				var node2 = new TreeViewNode2() { Content = "Inherited from TreeViewNode" };
+				Verify.AreEqual(node2.Content, node2.ToString());
+			});
+		}
+
+		[TestMethod]
+		public void TreeViewPendingSelectedNodesTest()
+		{
+			RunOnUIThread.Execute(() =>
+			{
+				var treeView = new TreeView();
+				treeView.SelectionMode = TreeViewSelectionMode.Multiple;
+
+				var node1 = new TreeViewNode() { Content = "Node1" };
+				var node2 = new TreeViewNode() { Content = "Node2" };
+				treeView.RootNodes.Add(node1);
+				treeView.RootNodes.Add(node2);
+				treeView.SelectedNodes.Add(node1);
+
+				Content = treeView;
+				Content.UpdateLayout();
+				Verify.AreEqual(true, IsMultiSelectCheckBoxChecked(treeView, node1));
+				Verify.AreEqual(false, IsMultiSelectCheckBoxChecked(treeView, node2));
+			});
+		}
+
+		//[TestMethod]
+		//public void VerifyVisualTree()
+		//{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)